src/build.d

```diff
@@ -756,6 +849,132 @@ void processEnvironment()
     flags["DFLAGS"] ~= dflags;
 }
 
+/// Setup environment for a C++ compiler
+version(Posix) void processEnvironmentCxx()
+{
+    import std.meta: AliasSeq;
+
+    env.getDefault("RES", srcDir ~ "/../res");
+
+    const cxxVersion = [env.getDefault("CXX", "c++"), "--version"].execute.output;
+
+    alias GCC = AliasSeq!("g++", "gcc", "Free Software");
+    alias CLANG = AliasSeq!("clang");
+    const kindIdx = cxxVersion.canFind(GCC, CLANG);
+
+    enforce(kindIdx, "Invalid CXX found: " ~ cxxVersion);
+    const kind = kindIdx <= GCC.length ? "g++" : "clang++";
+
+    env["CXX_KIND"] = kind;
+
+    string[] warnings;
+    if(env.getDefault("ENABLE_WARNINGS", "0") != "0")
+    {
+        warnings = [
+            "-Wall",
+            "-Wextra",
+            "-Werror",
+            "-Wno-attributes",
+            "-Wno-char-subscripts",
+            "-Wno-deprecated",
+            "-Wno-empty-body",
+            "-Wno-format",
+            "-Wno-missing-braces",
+            "-Wno-missing-field-initializers",
+            "-Wno-overloaded-virtual",
+            "-Wno-parentheses",
+            "-Wno-reorder",
+            "-Wno-return-type",
+            "-Wno-sign-compare",
+            "-Wno-strict-aliasing",
+            "-Wno-switch",
+            "-Wno-type-limits",
+            "-Wno-unknown-pragmas",
+            "-Wno-unused-function",
+            "-Wno-unused-label",
+            "-Wno-unused-parameter",
+            "-Wno-unused-value",
+            "-Wno-unused-variable"
+        ];
+
+        if(kind == "g++") {
+            warnings ~= [
+                "-Wno-logical-op",
+                "-Wno-narrowing",
+                "-Wno-unused-but-set-variable",
+                "-Wno-uninitialized",
+                "-Wno-class-memaccess",
+                "-Wno-implicit-fallthrough"
+            ];
+        }
+    }
+    else
+    {
+        warnings = [
+            "-Wno-deprecated",
+            "-Wstrict-aliasing",
+            "-Werror"
+        ];
+
+        if(kind == "clang++")
+            warnings ~= "-Wno-logical-op-parentheses";
+    }
+
+    string hostDmdVernum;
+    if(env["HOST_DMD_KIND"] == "dmd") // Why limit this to dmd?
+    {
+        enum redirect = Redirect.stdin | Redirect.stdout | Redirect.stderrToStdout;
+        auto pipes = pipeProcess([env["HOST_DMD_RUN"], "-o-", "-c", "-"], redirect);
+
+        pipes.stdin.writeln("pragma(msg, cast(int)__VERSION__);");
+        pipes.stdin.flush();
+        pipes.stdin.close();
+        //  wait(pipes.pid);
+
+        hostDmdVernum = pipes.stdout.byLine.front.idup;
+    }
+    else {
+        hostDmdVernum = "2";
+    }
+
+    auto cxxFlags = warnings ~ [
+        "-fno-exceptions",
+        "-fno-rtti",
+        "-D__pascal=",
+        "-DMARS=1",
+        "-DTARGET_" ~ env["OS"].toUpper ~ "=1",
+        "-DDM_TARGET_CPU_" ~ env["TARGET_CPU"] ~ "=1",
```

Comment:
You can get rid of `-D__pascal=`, `-DTARGET_XXX=1`, and `-DDM_TARGET_CPU_XXX=1` defines.  Just keep `-DMARS=1`

Reply:
Done